packages/EasyCodingStandard/packages/Configuration/src/ConflictingCheckerGuard.php

```diff
@@ -0,0 +1,55 @@
+<?php declare(strict_types=1);
+
+namespace Symplify\EasyCodingStandard\Configuration;
+
+use Symplify\EasyCodingStandard\Configuration\Exception\ConflictingCheckersLoadedException;
+
+final class ConflictingCheckerGuard
+{
+    /**
+     * These groups do the opposite of each other, e.g. Yoda vs NoYoda.
+     *
+     * @var string[][]
+     */
+    private static $conflictingCheckerGroups = [
+        [
+            'SlevomatCodingStandard\Sniffs\ControlStructures\DisallowYodaComparisonSniff',
+            'PhpCsFixer\Fixer\ControlStructure\YodaStyleFixer',
+        ], [
+            'PhpCsFixer\Fixer\Operator\UnaryOperatorSpacesFixer',
+            'PhpCsFixer\Fixer\Operator\NotOperatorWithSuccessorSpaceFixer',
+        ], [
+            'Symplify\CodingStandard\Sniffs\ControlStructures\NewClassSniff',
+            'PhpCsFixer\Fixer\Operator\NewWithBracesFixer',
+        ],
+    ];
+
+    /**
+     * @param mixed[] $checkers
+     */
+    public function processCheckers(array $checkers): void
+    {
+        foreach (self::$conflictingCheckerGroups as $viceVersaMatchingCheckerGroup) {
+            if (! $this->isMatch($checkers, $viceVersaMatchingCheckerGroup)) {
+                continue;
+            }
+
+            throw new ConflictingCheckersLoadedException(sprintf(
+                'Checkers "%s" mutually exclude each other. Use only one or exclude '
+                . 'the unwanted one in "parameters > exclude_checkers" in your config.',
+                implode('" and "', $viceVersaMatchingCheckerGroup)
+            ));
+        }
+    }
+
+    /**
+     * @param mixed[] $checkers
+     * @param string[] $matchingCheckerGroup
+     */
+    private function isMatch(array $checkers, array $matchingCheckerGroup): bool
+    {
+        $matchingCheckerGroupKeys = array_flip($matchingCheckerGroup);
+
+        return count(array_intersect_key($matchingCheckerGroupKeys, $checkers)) === count($matchingCheckerGroup);
```

Comment:
It is not easy to judge as we don't have such use case at the moment but I suspect that if there would be a group with 3 or more checkers we would want to throw an exception if any two of them are used. Which would mean this:

```php
return count(array_intersect_key($matchingCheckerGroupKeys, $checkers)) >= 2;
```

However then the exception would be wrong as it would contain unused checkers so it is slightly more complicated.

If you don't feel like changing the logic until we have such case I'm fine with that.

Reply:
I'd solve this when it comes. It'd be probably filtered out as duplicated checker in step before.